activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeClaimReleaseTest.java

```diff
@@ -0,0 +1,149 @@
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@ContextConfiguration
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class TaskRuntimeClaimReleaseTest {
+
+    private static String currentTaskId;
+    @Autowired
+    private TaskRuntime taskRuntime;
+    @Autowired
+    private TaskAdminRuntime taskAdminRuntime;
+    @Autowired
+    private SecurityManager securityManager;
+
+    @Test
+    @WithUserDetails(value = "garth", userDetailsServiceBeanName = "myUserDetailsService")
+    public void aCreateStandaloneTaskForGroup() {
+
+        String authenticatedUserId = securityManager.getAuthenticatedUserId();
+        Task standAloneTask = taskRuntime.create(TaskPayloadBuilder.create()
+                .withName("group task")
+                .withGroup("activitiTeam")
+                .build());
+
+        // the owner should be able to see the created task
+        Page<Task> tasks = taskRuntime.tasks(Pageable.of(0,
+                50));
+
+        assertThat(tasks.getContent()).hasSize(1);
+        Task task = tasks.getContent().get(0);
+
+        assertThat(task.getAssignee()).isNull();
+        assertThat(task.getStatus()).isEqualTo(Task.TaskStatus.CREATED);
+
+        currentTaskId = task.getId();
+
+    }
+
+
+    @Test
+    @WithUserDetails(value = "salaboy", userDetailsServiceBeanName = "myUserDetailsService")
+    public void bClaimAndRelease() {
+
+        String authenticatedUserId = securityManager.getAuthenticatedUserId();
+        Task claimedTask = taskRuntime.claim(TaskPayloadBuilder.claim().withTaskId(currentTaskId).build());
+        assertThat(claimedTask.getAssignee()).isEqualTo(authenticatedUserId);
+        assertThat(claimedTask.getStatus()).isEqualTo(Task.TaskStatus.ASSIGNED);
+
+        Task releasedTask = taskRuntime.release(TaskPayloadBuilder.release().withTaskId(claimedTask.getId()).build());
+        assertThat(releasedTask.getAssignee()).isNull();
+        assertThat(releasedTask.getStatus()).isEqualTo(Task.TaskStatus.CREATED);
+
+
+    }
+
+    @Test
+    @WithUserDetails(value = "garth", userDetailsServiceBeanName = "myUserDetailsService")
+    public void cCreateStandaloneTaskReleaseUnAuthorized() {
+
+        Task standAloneTask = taskRuntime.create(TaskPayloadBuilder.create()
+                .withName("group task")
+                .withGroup("activitiTeam")
+                .build());
+
+
+        assertThat(standAloneTask.getAssignee()).isNull();
+        assertThat(standAloneTask.getStatus()).isEqualTo(Task.TaskStatus.CREATED);
+
+        Exception e = null;
+        try {
+            // UnAuthorized release, task is not assigned
+            Task releasedTask = taskRuntime.release(TaskPayloadBuilder.release().withTaskId(standAloneTask.getId()).build());
```

Comment:
We can use the method `catchThrowAble` from AssertJ to replace this `try` `catch` block.

Reply:
@erdemedeiros please do.. I looked for it but.. I didn't found it in time.